src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+interface Product {
+  id: number;
+  title: string;
+  thumbnail: string;
+  description: string;
+  price: number;
+}
+
+function ProductDetails() {
+  const [product, setProduct] = useState<Product | null>(null);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [price, setPrice] = useState(0);
+  const { id } = useParams<{ id: string }>();
+
+  useEffect(() => {
+    fetch(`https://dummyjson.com/products/${id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        setProduct(data);
+        setTitle(data.title);
+        setDescription(data.description);
+        setPrice(data.price);
+      })
+      .catch((error) => console.log(error));
+  }, [id]);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setDescription(event.target.value);
+  };
+
+  const handlePriceChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPrice(parseFloat(event.target.value));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const updatedProduct = { ...product, name: title, description, price };
+    fetch(`https://dummyjson.com/products/${id}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(updatedProduct)
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setProduct(data);
+        setTitle(data.title);
+        setDescription(data.description);
+        setPrice(data.price);
+      })
+      .catch((error) => console.log(error));
+  };
+
+  if (!product) {
+    return <p>Loading product...</p>;
+  }
```

Comment:
while this might work, it is not proper because there may be some issue from the BE and the user would keep seeing a loading screen.

instead you should set loading to `true` before the API call, wait for the call to be finished, then set loading to `false`